### PR TITLE
Enable using script and nonextended keys

### PR DESCRIPTION
**Pull request by paweljakubas**
- [x] Support for key hashes and non-extended public key implementation using proper bits from cardano-addresses (rather than hack)
- [x] ApiCredential extension, also in swagger, golden regenerated in core unit testig
- [x] Added comprehensive instruction and more testing in integration suite

### Issue Number
adp-2114

### CODE REVIEW

File: cabal.project [hunk 1]
```diff
@@ -119,8 +119,8 @@ source-repository-package
 source-repository-package
     type: git
     location: https://github.com/input-output-hk/cardano-addresses
-    tag: b7273a5d3c21f1a003595ebf1e1f79c28cd72513
-    --sha256: 129r5kyiw10n2021bkdvnr270aiiwyq58h472d151ph0r7wpslgp
+    tag: 3172d30433b4cdd9beabe6b5a0a887351f5c0061
```
Comment (HeinrichApfelmus) [inline cabal.project]: I can't find this commit in the repo yet? 🤔 

https://github.com/input-output-hk/cardano-addresses/pull/202/commits/3172d30433b4cdd9beabe6b5a0a887351f5c0061

**Review by HeinrichApfelmus — APPROVED** [review]
Looks good to me! — I can find no obvious defects. 😊 

However, the commit hash for cardano-addresses does not exist yet? 🤔

**Comment by paweljakubas** [comment]
this is the commit from https://github.com/input-output-hk/cardano-addresses/pull/197
Of course #197 in cardano-addresses need to be merged first, and then cardano-addresses' commit bumped once again in this PR

**Pull request merged**